app/controllers/disclosure/ReplaceOrDeleteADisclosureController.scala

```diff
@@ -135,4 +141,10 @@ class ReplaceOrDeleteADisclosureController @Inject()(
           .withError(FormError("disclosureID", List("replaceOrDeleteADisclosure.error.disclosureID.notFound")))
     }
   }
+
+  private def replaceOrDelete(userAnswers: UserAnswers): Boolean = userAnswers.getBase(DisclosureTypePage) match {
+      case Some(Dac6rep) => true
+      case Some(Dac6del) => false
+      case _ => throw new RuntimeException("Disclosure type should only be replace or delete")
```

Comment:
Use ErrorHandler here to handle this?

Reply:
Added onErrorHandler to onSubmit and onPageLoad